src/Ng/Jobs/Feed2MonitoringJob.cs

```diff
@@ -0,0 +1,190 @@
+                            }
+                        }
+                        finally
+                        {
+                            if (!found)
+                            {
+                                // The package may have been scanned while we were parsing the feed, so check for its status again.
+                                var status = await _statusService.GetAsync(feedPackage, cancellationToken);
+                                if (status == null)
+                                {
+                                    // We can't find the package's status, so add it to the list of missing packages.
+                                    Logger.LogError($"Failed to find {feedPackage.Id} {feedPackage.Version}");
+                                    await _packageValidatorQueue.Add(
+                                        new PackageValidatorContext(feedPackage, null),
+                                        cancellationToken);
+                                    invalidPackages.Enqueue(feedPackage);
+                                }
+                                else
+                                {
+                                    // We found the package's status in our second check.
+                                    found = true;
+                                }
+                            }
+                        }
+                    }
+                }
+                else
+                {
+                    var page = Interlocked.Increment(ref _pagesProcessed);
+
+                    V2FeedPage nextFeedPage = await RepeatUntilSuccessOrMaxFailures(() => _feedParser.GetPackagesPageAsync("", new SearchFilter(true), page * _pageSize, _pageSize, NullLogger.Instance, cancellationToken));
+
+                    if (nextFeedPage != null && nextFeedPage.Items.Count > 0)
+                    {
+                        foreach (var unprocessedPackage in nextFeedPage.Items)
+                        {
+                            queue.Enqueue(unprocessedPackage);
+                        }
+                    }
+                    else
+                    {
+                        shouldContinue = false;
+                    }
+                }
+            }
+        }
+
+        private async Task<HashSet<FeedPackageIdentity>> GetValidPackages(CancellationToken cancellationToken)
+        {
+            var validPackagesList = await _statusService.ListAsync(cancellationToken);
+            return new HashSet<FeedPackageIdentity>(validPackagesList.Select(p => p.Package).ToList());
+        }
+
+        private static async Task<T> RepeatUntilSuccessOrMaxFailures<T>(Func<Task<T>> action) where T : class
+        {
+            var requestSuccessful = false;
+            var failures = 0;
+
+            T result = null;
+            while (!requestSuccessful || failures > _maxRequestFailures)
+            {
+                try
+                {
+                    result = await action();
```

Comment:
>result [](start = 20, length = 6)

Nit: you could just return the result of the action directly, get rid of the `requestSuccessful` variable, and return `null` on line 187.